src/mesh/mesh.h

```diff
@@ -357,9 +342,58 @@ class Mesh {
     /// Maximal distance of observe point from Mesh relative to its size
     double global_observe_radius() const;
 
-    /// Number of elements read from input.
-    unsigned int n_all_input_elements_;
+    /// Initialize element_vec_, set size and reset counters of boundary and bulk elements.
+    void init_element_vector(unsigned int size);
 
+    /// Returns range (of indexes in element_vector_) of boundary or bulk elements
+    inline pair<unsigned int, unsigned int> get_element_range(bool boundary=false) const {
```

Comment:
bulk_elements(), boundary_elements()
```
return  make_iterator_range(make_iter(ElementAccessor(mesh,0), make_iter(ElementAccessor(mesh, n_bulk_elements))
```

Reply:
Try to implement own Range class (having begin, end and size methods). Test if such a class can be used in the for statement.